flocker/node/agents/blockdevice.py

```diff
@@ -53,6 +59,125 @@ class UnattachedVolume(VolumeException):
     """
 
 
+DATASET = Field(
+    u"dataset",
+    lambda dataset: dataset.dataset_id,
+    u"The unique identifier of a dataset."
+)
+
+DATASET_ID = Field(
+    u"dataset_id",
+    lambda dataset_id: unicode(dataset_id),
+    u"The unique identifier of a dataset."
+)
+
+MOUNTPOINT = Field(
+    u"mountpoint",
+    lambda path: path.path,
+    u"The absolute path to the location on the node where the dataset will be "
+    u"mounted.",
+)
+
+DEVICE_PATH = Field(
+    u"block_device_path",
+    lambda path: path.path,
+    u"The absolute path to the block device file on the node where the "
+    u"dataset is attached.",
+)
+
+BLOCK_DEVICE_ID = Field(
+    u"block_device_id",
+    lambda id: unicode(id),
+    u"The unique identifier if the underlying block device."
+)
+
+BLOCK_DEVICE_SIZE = Field(
+    u"block_device_size",
+    lambda size: size,
+    u"The size of the underlying block device."
+)
+
+BLOCK_DEVICE_HOST = Field(
+    u"block_device_host",
+    lambda host: host,
+    u"The host to which the underlying block device is attached."
+)
+
+CREATE_BLOCK_DEVICE_DATASET = ActionType(
+    u"agent:blockdevice:create",
+    [DATASET, MOUNTPOINT],
+    [DEVICE_PATH, BLOCK_DEVICE_ID, DATASET_ID, BLOCK_DEVICE_SIZE,
+     BLOCK_DEVICE_HOST],
+    u"A block-device-backed dataset is being created.",
+)
+
+
+@implementer(IStateChange)
+class CreateBlockDeviceDataset(PRecord):
+    """
+    An operation to create a new dataset on a newly created volume with a newly
+    initialized filesystem.
+
+    :ivar Dataset dataset: The dataset for which to create a block device.
+    :ivar FilePath mountpoint: The path at which to mount the created device.
+    :ivar Logger logger: An Eliot ``Logger``.
+    """
+    dataset = field()
+    mountpoint = field(mandatory=True)
+
+    logger = Logger()
+
+    def run(self, deployer):
+        """
+        Create a block device, attach it to the host of the supplied
```

Comment:
This is a little misleading, since the code below only works if you're running this on the host where it will be attached. So maybe say "local host".